src/metabase/pulse/render/common.clj

```diff
@@ -25,55 +28,110 @@
   Object
   (toString [_] num-str))
 
+(defn- strip-trailing-zeroes
+  [num-as-string decimal]
+  (if (str/includes? num-as-string (str decimal))
+    (let [pattern (re-pattern (str/escape (str decimal \$) {\. "\\."}))]
+      (-> num-as-string
+          (str/split #"0+$")
+          first
+          (str/split pattern)
+          first))
+    num-as-string))
+
+(defn- digits-after-decimal
+  ([value] (digits-after-decimal value "."))
+  ([value decimal]
+   (if (zero? value)
+     0
+     (let [val-string (-> (condp = (type value)
+                            java.math.BigDecimal (.toPlainString ^BigDecimal value)
+                            java.lang.Double (format "%.20f" value)
+                            java.lang.Float (format "%.20f" value)
+                            (str value))
+                          (strip-trailing-zeroes (str decimal)))
+           [_n d] (str/split val-string #"[^\d*]")]
+       (count d)))))
+
+(defn- sig-figs-after-decimal
+  [value decimal]
+  (if (zero? value)
+    0
```

Comment:
This says that `0.000000000000` has no sig-figs after decimal. Is this correct? Just wondering about that case.

Reply:
ah i missed this returned `0` here. My brain was operating as if it returned `value`. Interested to see you confirm but i bet this is fine then.

Reply:
For sure 0.0000 has zero significant figures after the decimal, so this is correct behavior.

Reply:
I'm not sure I agree with that. `0.0001` and `0.0000` both have the same number of significant figures. Our UI collapses all zero numbers to just `0` but that's independent of a notion of significant figures. https://en.wikipedia.org/wiki/Significant_figures#Rules_to_identify_significant_figures_in_a_number

> Zeros to the right of the last non-zero digit ([trailing zeros](https://en.wikipedia.org/wiki/Trailing_zero)) in a number with the decimal point are significant if they are within the measurement or reporting resolution.

    1.200 has four significant figures (1, 2, 0, and 0) if they are allowed by the measurement resolution.
    0.0980 has three significant digits (9, 8, and the last zero) if they are within the measurement resolution.
    120.000 consists of six significant figures (1, 2, and the four subsequent zeroes).

Reply:
I don't see how this is solvable if `value` is coming in as a number. To Dan's point, `0.00` has two sig figs and `0.000` has three, but of course Clojure will treat any float representing 0 as the same thing.

I also have concerns about floating point precision (e.g. if you get `0.0999999` passed in instead of `0.1`—check out what `(reduce + (repeat 10 0.1))` returns) but haven't figured out if they're legit yet.